src/app/shared/classes/test-flow-base-pages/test-report/test-report-base-page.ts

```diff
@@ -41,7 +129,209 @@ export abstract class TestReportBasePageComponent extends PracticeableBasePageCo
+      case TestCategory.F:
+      case TestCategory.G:
+      case TestCategory.H:
+      case TestCategory.K:
+        return getTestRequirementsCatHome(testData) as CatHUniqueTypes.TestRequirements;
+      default:
+        return getTestRequirementsCatB(testData) as CatBUniqueTypes.TestRequirements;
+    }
+  }
+
+  hasManoeuvreBeenCompleted(data: TestDataUnion, category: CategoryCode) {
+    switch (category) {
+      case TestCategory.B:
+        return hasManoeuvreBeenCompletedCatB(data as CatBUniqueTypes.TestData);
+      case TestCategory.BE:
+        return hasManoeuvreBeenCompletedCatBE(data as CatBEUniqueTypes.TestData);
+      case TestCategory.C:
+      case TestCategory.C1:
+      case TestCategory.C1E:
+        return hasManoeuvreBeenCompletedCatC(data as CatCUniqueTypes.TestData);
+      case TestCategory.D:
+      case TestCategory.D1:
+      case TestCategory.DE:
+      case TestCategory.D1E:
+        return hasManoeuvreBeenCompletedCatD(data as CatDUniqueTypes.TestData);
+      case TestCategory.F:
+      case TestCategory.G:
+      case TestCategory.H:
+      case TestCategory.K:
+        return hasManoeuvreBeenCompletedCatHomeTest(data as CatHUniqueTypes.TestData);
+      default:
+        return null;
+    }
+  }
+
+  async ionViewWillEnter(): Promise<void> {
+    // ionViewWillEnter lifecylce event used to ensure screen orientation is correct before page transition
+    if (super.isIos() && this.isPracticeMode) {
+      await this.screenOrientation.lock(
+        this.screenOrientation.ORIENTATIONS.PORTRAIT_PRIMARY,
+      );
+      await this.insomnia.keepAwake();
+      this.statusBar.hide();
+    }
+  }
+
+  ionViewDidEnter(): void {
+    // it is possible that we come back to the page from the terminate screen
+    // so need to re-establish the subscription if it doesn't exists or is closed
+    if (!this.subscription || this.subscription.closed) {
+      this.setupSubscription();
+    }
+    this.store$.dispatch(TestReportViewDidEnter());
+  }
+
+  ionViewWillLeave() {
+    if (super.isIos() && this.isPracticeMode) {
+      this.statusBar.show();
+    }
+  }
+
+  toggleReportOverlay(): void {
+    this.displayOverlay = !this.displayOverlay;
+  }
+
+  setupSubscription() {
+    const {
+      candidateUntitledName$,
+      isRemoveFaultMode$,
+      isSeriousMode$,
+      isDangerousMode$,
+      manoeuvres$,
+      testData$,
+    } = this.commonPageState;
+
+    this.subscription = merge(
+      candidateUntitledName$,
+      isRemoveFaultMode$.pipe(map((result) => (this.isRemoveFaultMode = result))),
+      isSeriousMode$.pipe(map((result) => (this.isSeriousMode = result))),
+      isDangerousMode$.pipe(map((result) => (this.isDangerousMode = result))),
+      manoeuvres$.pipe(map((result) => (this.manoeuvresCompleted = result))),
+      testData$.pipe(map((data) => {
+        this.isTestReportValid = this.testReportValidatorProvider.isTestReportValid(data, TestCategory.B);
```

Comment:
Can we grab the TestCategory here instead of hardcoding the value?

Reply:
sorry managed to miss this will update

Reply:
updated